### PR TITLE
Rough draft of connection to firebase

### PR DESCRIPTION
**Pull request by granttitus**

### CODE REVIEW

File: src/game/Actions/Actions.tsx [hunk 1]
```diff
@@ -3,29 +3,29 @@ import { shuffle } from "lodash"
...
+    getCardInDiscardById,
+    getCardInDrawPileById,
+    getCardInArchiveById,
     getPlayerByName,
     removeCreature,
     removeArtifact,
     removeUpgrade,
     removeCardFromHand,
-    removeCardByID,
+    removeCardById,
```
Comment (granttitus) [inline src/game/Actions/Actions.tsx]: Changed `ID` to `Id` to be consistent across all our code

File: src/game/Utils.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import Phaser from "phaser"
+
+const cardNameToUrl = (name: string) => `https://keyforge-card-images.s3-us-west-2.amazonaws.com/card-imgs/${name}.png`
+
+export const preloadCardsInPhaser = (scene: Phaser.Scene, cards: {id: string}[]) => {
+    cards.forEach((card: {id: string}) => {
+        console.log(card)
+        let id = card.id
+        if (card.id === "arise!")
+            id = "arise"
```
Comment (granttitus) [inline src/game/Utils.tsx]: The `!` in arise needs to be removed in the URL. I'll remove the `console.log` after I take a look at the entire card list and make sure other characters aren't an issue.

File: src/game/types/Player.tsx [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
+    amber: number
+    chains: number
+    keys: number
+    creatures: Creature[]
+    artifacts: Artifact[]
+    hand: CardNotInPlay[]
+    drawPile: CardNotInPlay[]
+    discardPile: CardNotInPlay[]
+    archivePile: CardNotInPlay[]
+    purgePile: CardNotInPlay[]
```
Comment (granttitus) [inline src/game/types/Player.tsx]: Need to delete this file and use the other Player type.

File: src/game/Actions/Actions.tsx (hunk 1 above)
Comment (CorayThan) [reply src/game/Actions/Actions.tsx]: Ah, cool. I didn't even think about that. I typically write id in camel case.

File: src/game/Utils.tsx (hunk 2 above)
Comment (CorayThan) [reply src/game/Utils.tsx]: oh weird, my regex didn't match `!`? I'll need to fix that then

Comment (CorayThan) [reply src/game/Utils.tsx]: Oh, looks like you deleted the regex?

Comment (granttitus) [reply src/game/Utils.tsx]: That was a mistake. I'll add it back.

**Pull request merged**